scipy/_lib/tests/test__util.py

```diff
@@ -85,37 +85,41 @@ def test_serial(self):
             p = MapWrapper(0)
 
     def test_parallel(self):
-        with MapWrapper(2) as p:
-            out = p(np.sin, self.input)
-            assert_equal(list(out), self.output)
-
-            assert_(p._own_pool is True)
-            assert_(isinstance(p.pool, PWL))
-            assert_(p._mapfunc is not None)
-
-        # the context manager should've closed the internal pool
-        # check that it has by asking it to calculate again.
-        with assert_raises(Exception) as excinfo:
-            p(np.sin, self.input)
-
-        # on py27 an AssertionError is raised, on >py27 it's a ValueError
-        err_type = excinfo.type
-        assert_((err_type is ValueError) or (err_type is AssertionError))
-
-        # can also set a PoolWrapper up with a map-like callable instance
-        try:
-            p = Pool(2)
-            q = MapWrapper(p.map)
-
-            assert_(q._own_pool is False)
-            q.close()
-
-            # closing the PoolWrapper shouldn't close the internal pool
-            # because it didn't create it
-            out = p.map(np.sin, self.input)
-            assert_equal(list(out), self.output)
-        finally:
-            p.close()
+        # The line __name__ == '__main__' is needed for platforms that
+        # uses spawn() instead of fork() to start new processes, to get
+        # out of infinite recursion.
+        if __name__ == '__main__':
```

Comment:
Under what conditions is this condition true?  Have you, in effect, simply commented-out the test, for all platforms?

Reply:
This condition is true only for the first time this instance is created, then for other processes the name is not main anymore. Hence it blocks the recursion.

Reply:
My understanding is that `__name__` is `__main__` when the file is run as a script.  Does `pytest` run the file as a script?  I just added a couple lines with `assert_(False)` to the test, and it still passed.  I.e. the code in the `with` context never ran.